src/hypergraph.jl

```diff
@@ -5,85 +5,122 @@ A hypergraph storing information about vertices and hyperedges.
             v_meta, he_meta)
 end
 
 
-Hypergraph{T}(n, k) where {T<:Real} =  Hypergraph{T,Nothing,Nothing}(n, k)
+Hypergraph{T,V,E}(n::Integer, k::Integer) where {T<:Real, V, E} = Hypergraph{T,V,E,Dict{Int,T}}(n, k)
+
+Hypergraph{T,V}(n::Integer, k::Integer) where {T<:Real, V, E} = Hypergraph{T,V,Nothing,Dict{Int,T}}(n, k)
+
+Hypergraph{T}(n::Integer, k::Integer) where {T<:Real} =  Hypergraph{T,Nothing,Nothing,Dict{Int,T}}(n, k)
 
-Hypergraph{T,V}(n, k;
-              v_meta=Vector{Union{V,Nothing}}(nothing, size(m,1))
-             ) where {T<:Real, V} = Hypergraph{T,V,Nothing}(n, k; v_meta=v_meta)
+Hypergraph(n::Integer, k::Integer) =  Hypergraph{Bool,Nothing,Nothing,Dict{Int,Bool}}(n, k)
 
-function Hypergraph{V, E}(m::AbstractMatrix{Union{T, Nothing}};
-                        v_meta=Vector{Union{V,Nothing}}(nothing, size(m,1)),
-                        he_meta=Vector{Union{E,Nothing}}(nothing, size(m,2))
-                        ) where {T<:Real, V, E}
+function Hypergraph{T,V,E,D}(m::AbstractMatrix{Union{T, Nothing}};
+                        v_meta::Vector{Union{Nothing,V}}=Vector{Union{Nothing,V}}(nothing, size(m,1)),
+                        he_meta::Vector{Union{Nothing,E}}=Vector{Union{Nothing,E}}(nothing, size(m,2))
+                        ) where {T<:Real,V,E,D<:AbstractDict{Int,T}}
+    @assert length(v_meta) == size(m,1)
+    @assert length(he_meta) == size(m,2)
     n, k = size(m)
-    h = Hypergraph{T, V, E}(n, k, v_meta, he_meta)
+    h = Hypergraph{T,V,E,D}(n, k, v_meta, he_meta)
     h .= m
     h
 end
 
-Hypergraph{V}(m::AbstractMatrix{Union{T, Nothing}};
-            v_meta=Vector{Union{V,Nothing}}(nothing, size(m,1))) where {T<:Real, V} =
-    Hypergraph{V, Nothing}(m; v_meta=v_meta)
-Hypergraph(m::AbstractMatrix{Union{T, Nothing}}) where {T<:Real} =
-    Hypergraph{Nothing, Nothing}(m)
+function Hypergraph{T,V,E}(m::AbstractMatrix{Union{T, Nothing}};
+                        v_meta::Vector{Union{Nothing,V}}=Vector{Union{Nothing,V}}(nothing, size(m,1)),
+                        he_meta::Vector{Union{Nothing,E}}=Vector{Union{Nothing,E}}(nothing, size(m,2))
+                        ) where {T<:Real,V,E}
+    Hypergraph{T,V,E,Dict{Int,T}}(m;v_meta=v_meta,he_meta=he_meta)
+end
+
+function Hypergraph{T,V}(m::AbstractMatrix{Union{T, Nothing}};
+                        v_meta::Vector{Union{Nothing,V}}=Vector{Union{Nothing,V}}(nothing, size(m,1))
+                        ) where {T<:Real,V}
+    Hypergraph{T,V,Nothing,Dict{Int,T}}(m;v_meta=v_meta)
+end
+
+function Hypergraph{T}(m::AbstractMatrix{Union{T, Nothing}}) where {T<:Real}
+    Hypergraph{T,Nothing,Nothing,Dict{Int,T}}(m)
+end
+
+function Hypergraph(m::AbstractMatrix{Union{T, Nothing}}) where {T<:Real}
+    Hypergraph{T,Nothing,Nothing,Dict{Int,T}}(m)
+end
+
 
 function Hypergraph(g::LightGraphs.Graph)
-    h = Hypergraph{Bool}(maximum(vertices(g)), ne(g))
+    h = Hypergraph{Bool,Nothing,Nothing,SortedDict{Int,Bool}}(maximum(vertices(g)), ne(g))
```

Comment:
why do you use `SortedDict` here?

Reply:
`LightGraphs.SimpleGraph` keeps the adjacency list in a deterministic sorted order. Hence the equivalent will be `SortedDict`